### PR TITLE
Migrate to github actions

### PR DESCRIPTION
**Pull request by krolol**

### CODE REVIEW

File: .github/workflows/dev.yaml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      - name: Configure AWS credentials for push images
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+      - name: Set up Docker Buildx
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml:28]: Why Build**x**?

File: .github/workflows/dev.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
+name: Release to dev
+on:
+  push:
+    branches:
+      - dev-cere
+  workflow_dispatch:
+
+jobs:
+  build:
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Let's rename to `build_backend`

File: .github/workflows/dev.yaml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        with:
+          context: ./
+          file: ./backend/docker/crawler/Dockerfile
+          push: true
+          no-cache: true
+          build-args: |
+            NPM_TOKEN=${{ secrets.NPM_TOKEN }}
+          tags: ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ steps.version.outputs.version }}
+
+  build_cerestats_frontend:
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Let's rename to `build_frontend`

File: .github/workflows/dev.yaml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          yarn install
+      - name: Build
+        run: yarn workspace frontend generate
+      - name: Prepare artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: frontend/dist
+
+  deploy:
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Let's rename to `deploy_backend`?

File: .github/workflows/dev.yaml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    runs-on: [self-hosted, cere-network-dev-deployer]
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml:85]: Let's follow the same format everywhere. In line 58 you have spaces after and before square brackets.

File: .github/workflows/dev.yaml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+          ENVIRONMENT: dev
+        run: |
+          helm upgrade --install ${HELM_RELEASE} ${HELM_REPO_PATH} \
+            -f ${HELM_REPO_PATH}/envs/${ENVIRONMENT}.yaml \
+            --set image.tag=${{ needs.build.outputs.version }} \
+            --debug --atomic --cleanup-on-fail \
+            --timeout 5m \
+            -n ${NAMESPACE}
+
+  deploy_cerestats_frontend:
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Let's rename to `deploy_frontend`

File: .github/workflows/stage.yaml [hunk 7]
```diff
@@ -0,0 +1,155 @@
+name: Release to stage
+on:
+  push:
+    branches:
+      - 'releases/**'
```
Comment (AndreiNavoichyk) [inline .github/workflows/stage.yaml]: Rename to `release/**`

File: .github/workflows/prod.yaml [hunk 8]
```diff
@@ -0,0 +1,154 @@
+name: Release to prod
+on:
+  push:
+    branches:
+      - master-cere
+  workflow_dispatch:
```
Comment (AndreiNavoichyk) [inline .github/workflows/prod.yaml]: We need it only for DEV

File: .github/workflows/stage.yaml [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+      - 'hotfix/**'
+  workflow_dispatch:
```
Comment (AndreiNavoichyk) [inline .github/workflows/stage.yaml]: We need it only for DEV

**Review by AndreiNavoichyk — COMMENTED** [review]
@krolol 
Please, apply fixes from DEV to all envs.
Please, do not resolve comments. Just reply and I will close after checking.

File: .github/workflows/dev.yaml (hunk 1 above)
Comment (krolol) [reply .github/workflows/dev.yaml:28]: docker/build-push-action@v2 not works without Buildx

File: .github/workflows/dev.yaml [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+      - name: Process version
+        id: version
+        run: |
+          echo "::set-output name=version::dev-${{ github.run_attempt}}-$(git rev-parse --short HEAD)"
+      - name: Configure AWS credentials for push images
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/prod.yaml [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+      - name: Process version
+        id: version
+        run: |
+          echo "::set-output name=version::prd-${{ github.run_attempt}}-$(git rev-parse --short HEAD)"
+      - name: Configure AWS credentials for push images
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (AndreiNavoichyk) [inline .github/workflows/prod.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/stage.yaml [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+      - name: Process version
+        id: version
+        run: |
+          echo "::set-output name=version::stg-${{ github.run_attempt}}-$(git rev-parse --short HEAD)"
+      - name: Configure AWS credentials for push images
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (AndreiNavoichyk) [inline .github/workflows/stage.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/dev.yaml [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+          name: dist
+          path: frontend/dist
+
+  deploy_backend:
+    runs-on: [self-hosted, cere-network-dev-deployer]
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: "cere-io/infra-k8s-deployment"
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/prod.yaml [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+          name: dist
+          path: frontend/dist
+
+  deploy_backend:
+    runs-on: [self-hosted, cere-network-prd-deployer]
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: "cere-io/infra-k8s-deployment"
```
Comment (AndreiNavoichyk) [inline .github/workflows/prod.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/stage.yaml [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+          name: dist
+          path: frontend/dist
+
+  deploy_backend:
+    runs-on: [self-hosted, cere-network-stg-deployer]
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: "cere-io/infra-k8s-deployment"
```
Comment (AndreiNavoichyk) [inline .github/workflows/stage.yaml]: Should it be moved to ORG_SECRETS?

File: .github/workflows/dev.yaml [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+        run: |
+          helm upgrade --install ${HELM_RELEASE} ${HELM_REPO_PATH} \
+            -f ${HELM_REPO_PATH}/envs/${ENVIRONMENT}.yaml \
+            --set image.tag=${{ needs.build.outputs.version }} \
+            --debug --atomic --cleanup-on-fail \
+            --timeout 5m \
+            -n ${NAMESPACE}
+
+  deploy_frontend:
+    runs-on: [ self-hosted, cere-network-large-workers ]
```
Comment (AndreiNavoichyk) [inline .github/workflows/dev.yaml]: Please, fix formatting

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**